GraphBase/src/algorithm/Traversal.java

```diff
@@ -0,0 +1,73 @@
+package algorithm;
+
+import graph.Graph;
+import graph.Vertex;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Queue;
+import java.util.LinkedList;
+import java.util.Stack;
+
+public class Traversal {
+
+  public ArrayList<Integer> BFS(Graph graph, Integer entryId) {
+    Set<Integer> visitMark = new HashSet<Integer>();
+    Queue<Integer> visitQueue = new LinkedList<Integer>();
+    Map<Integer, Vertex> vertices = graph.getVertices();
+    ArrayList<Integer> visitOrder = new ArrayList<>();
+    if (!vertices.containsKey(entryId)) {
+      throw new NoSuchElementException("The entry vertex is not existed");
+    }
+    visitQueue.add(entryId);
+    while (!visitQueue.isEmpty()) {
```

Comment:
Take care. If the graph is disconnected, you will only visit the connected component of the entry node.